validation/biogeochemistry/simple_plankton_continuous_form_biogeochemistry.jl

```diff
@@ -0,0 +1,182 @@
+using Oceananigans, Printf, KernelAbstractions
+using Oceananigans.Units: minutes, hour, hours, day, days
+using Oceananigans.Biogeochemistry: AbstractContinuousFormBiogeochemistry, BasicBiogeochemistry, all_fields_present
+using Oceananigans.Grids: znode
+using Oceananigans.Forcings: maybe_constant_field
+using Oceananigans.Architectures: device, architecture
+using Oceananigans.Utils: launch!
+using Oceananigans.Advection: CenteredSecondOrder
+using Oceananigans.Fields: Field, TracerFields, CenterField
+
+import Oceananigans.Biogeochemistry:
+       required_biogeochemical_tracers,
+       required_biogeochemical_auxiliary_fields,
+       biogeochemical_drift_velocity,
+       biogeochemical_advection_scheme,
+       biogeochemical_auxiliary_fieilds,
+       update_biogeochemical_state!
+
+struct SimplePlanktonGrowthDeath{FT, W, SP, A, P} <: AbstractContinuousFormBiogeochemistry
+    growth_rate :: FT
+    light_limit :: FT
+    mortality_rate :: FT
+    water_light_attenuation_coefficient :: FT
+    phytoplankton_light_attenuation_coefficient :: FT
+    phytoplankton_light_attenuation_exponent :: FT
+    sinking_velocity :: W
+    surface_PAR :: SP
+    advection_scheme :: A
+    PAR :: P
+end
+
+@inline default_surface_radiation(t) = 100 * max(0, sin(t * π / 12hours))
+
+function SimplePlanktonGrowthDeath(FT=Float64; grid,
+                                   growth_rate = 1/day,
+                                   light_limit = 3.5,
+                                   mortality_rate = 0.3/day,
+                                   sinking_velocity = 0.0,
+                                   water_light_attenuation_coefficient = 0.01,
+                                   phytoplankton_light_attenuation_coefficient = 0.3,
+                                   phytoplankton_light_attenuation_exponent = 0.6,
+                                   surface_PAR = default_surface_radiation,
+                                   advection_scheme = nothing)
+
+    if sinking_velocity != 0
+        advection_scheme = CenteredSecondOrder()
+    end
+
+    u, v, w = maybe_constant_field.((0.0, 0.0, - sinking_velocity))
```

Comment:
In general we should try to avoid writing `0.0` because it assumes / promotes to double precision. `0` would work fine here I think.